django_modern_rest/internal/middleware_wrapper.py

```diff
@@ -8,34 +8,38 @@
 if TYPE_CHECKING:
     from django_modern_rest.response import ResponseDescription
 
-TypeT = TypeVar('TypeT', bound=type[Any])
+_TypeT = TypeVar('_TypeT', bound=type[Any])
 _CallableAny: TypeAlias = Callable[..., Any]
 MiddlewareDecorator: TypeAlias = Callable[[_CallableAny], _CallableAny]
 ResponseConverter: TypeAlias = Callable[[HttpResponse], HttpResponse]
-_ConverterSpec: TypeAlias = tuple['ResponseDescription', ResponseConverter]
+_ConverterSpec: TypeAlias = tuple[
+    list['ResponseDescription'],
+    ResponseConverter,
+]
 _ViewDecorator: TypeAlias = Callable[[_CallableAny], _CallableAny]
 
 
 @dataclass(frozen=True, slots=True, kw_only=True)
 class DecoratorWithResponses:
     """Type for decorator with responses attribute."""
 
-    decorator: Callable[[Any], Any]
+    decorator: Callable[[_TypeT], _TypeT]  # pyright: ignore[reportGeneralTypeIssues]
     responses: list['ResponseDescription']
 
-    def __call__(self, klass: TypeT) -> TypeT:
+    def __call__(self, klass: _TypeT) -> _TypeT:
         """Apply the decorator to the class."""
-        return self.decorator(klass)  # type: ignore[no-any-return]
+        return self.decorator(klass)  # pyright: ignore[reportReturnType]
 
 
 def apply_converter(
     response: HttpResponse,
     converter: _ConverterSpec,
 ) -> HttpResponse:
     """Apply response converter based on status code matching."""
-    response_desc, converter_func = converter
-    if response.status_code == response_desc.status_code:
-        return converter_func(response)
+    response_descs, converter_func = converter
+    for response_desc in response_descs:
```

Comment:
Nope, we can't do that. Too slow, we need O(1), not O(N). Convert to `dict` first. Plus, add `# TODO` comment to validate this responses definition.

See `validation.py`